src/js/components/DataTable/Header.js

```diff
@@ -297,27 +298,16 @@ const Header = forwardRef(
                   </Box>
                 );
               }
-              const pin = [];
-              if (tablePin) pin.push('top');
-              if (columnPin) pin.push('left');
+              const cellPin = [...pin];
+              if (columnPin) cellPin.push('left');
```

Comment:
wouldn't `columnPin` always has a value of at least an empty array? Should this be refined to asking about the array's length?

Reply:
columnPin is a boolean property passed in at line 296 of DataTable.js